### ISSUE TITLE
Feature request: Quick toggle bandwidth limits

### ISSUE THREAD

**Issue by benshep** [canton7/SyncTrayzor#328]
Would it be possible to add a tray menu item to quickly toggle bandwidth limiting? I often want to limit the bandwidth used by Syncthing so that other apps (and devices) can use it, but it would be nice to be able to quickly toggle it on/off instead of having to go into the web interface. (I realise that it means a restart.)

**Comment by scienmind** [comment]
I'd love to see something like that as well, over the past few years all kinds of torrent clients started to do this (qbittorrent, transmission, etc.) and it is indeed very useful to free resources "in a quick single click".

However, if restart is needed for it to apply, it'll be much less useful (at least for me :wink: ) . 
ST's restart is a heavy task (once you have many folders...), that i'd rather avoid. And if the toggle would depend on it, i'd just shutdown ST for the time i need the resources instead...
But that's more of a rant on (awesome nonetheless) syncthing, and it's subsystems. It may be improved one day.

As far as having this toggle in SyncTrayzor - i don't see why not :)

**Comment by canton7** [comment]
> As far as having this toggle in SyncTrayzor - i don't see why not :)

Yeah, you're not the one working on it in all of your lunch breaks for a week 😉

**Comment by scienmind** [comment]
Oh I know that 😉

As so do you, that all those are "ideas suggestions" that you are free to dismiss (and we are free to submit a pr during our lunch breaks)

Anyways, the intention was just to express my opinion, sorry if it came up arrogant...

**Comment by canton7** [comment]
To elaborate, this a bit harder than it looks:

1. I need to be able to modify the config and post it back. I don't do this currently, as being able to consume and modify a configuration format while still allowing it to change in the future (and remaining compatible with those unknown future changes) is hard.
2. New settings inputs to specify the bandwidth limits you want to apply.
3. Noticing when the user edits the bandwidth limits from within Syncthing, and doing something sensible. If they apply limits within Syncthing, presumably we display "bandwidth limits on"? If bandwidth limits are on and the user changes them from within Syncthing, presumably we update our local record of what the bandwidth limits should be?
4. To avoid cluttering the context menu on the tray icon, presumably the option to enable/disable the bandwidth limit should only be displayed if bandwidth limits have been configured.
5. Should the tray icon show something? Otherwise it might not be obvious why things are going so slow.

**Comment by benshep** [comment]
Hmm, I hadn't thought about all that. Maybe it would be worth asking the
Syncthing developers, it might be easier to implement at that level.
